go-controller/pkg/ovn/common.go

```diff
@@ -2,47 +2,67 @@ package ovn
 
 import (
 	"fmt"
-	goovn "github.com/ebay/go-ovn"
-	util "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
 	"k8s.io/klog/v2"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
 // hash the provided input to make it a valid portGroup name.
 func hashedPortGroup(s string) string {
 	return util.HashForOVN(s)
 }
 
-func createPortGroup(ovnNBClient goovn.Client, name string, hashName string) (string, error) {
+func createPortGroup(nbClient libovsdbclient.Client, name string, hashName string) (string, error) {
 	klog.V(5).Infof("createPortGroup with %s", name)
 	externalIds := map[string]string{"name": name}
-	cmd, err := ovnNBClient.PortGroupAdd(hashName, nil, externalIds)
-	if err == nil {
-		if err = ovnNBClient.Execute(cmd); err != nil {
-			return "", fmt.Errorf("execute error for add port group: %s, %v", name, err)
-		}
-	} else if err != goovn.ErrorExist {
-		// Ignore goovn.ErrorExist to implement "--may-exist" behavior
-		return "", fmt.Errorf("add error for port group: %s, %v", name, err)
-	}
-
-	pg, err := ovnNBClient.PortGroupGet(hashName)
-	if err == nil {
-		return pg.UUID, nil
-	} else {
-		return "", fmt.Errorf("failed to get port group UUID: %s, %v", name, err)
+
+	pg := nbdb.PortGroup{
+		Name:        hashName,
+		ExternalIDs: externalIds,
+	}
+	ops, err := nbClient.Create(&pg)
```

Comment:
I really think we should have wrappers for operations here, to collapse all this down into one line like so:

```
	results, err := CreatePortGroup(&nbdb.PortGroup{
		Name:        hashName,
		ExternalIDs: externalIds,
	})
	if err != nil {
		return "", err
	}
	return results[0].UIUD.GoUUID, nil
```

and the same for Delete, Update/Mutate, etc. Open-coding the <Operation> + Transact + CheckOperationResults every single time is error prone, ugly, and not quite as testable right?

Reply:
Or, it would be enough for me to have a common call that combines these two things:

```
	results, err := nbClient.Transact(ops...)
	if err != nil {
		return "", fmt.Errorf("Error creating port group with ops %+v: %v", ops, err)
	}	opErrors, err := libovsdb.CheckOperationResults(results, ops)
	if err != nil {
		return "", fmt.Errorf("Error creating port group with results %+v and errors %+v: %v", results, opErrors, err)
	}
```

since we seem to call these together very often.